pkg/url/url.go

```diff
@@ -18,15 +18,19 @@ import (
 	"fmt"
 
 	"istio.io/istio/operator/version"
+	buildversion "istio.io/pkg/version"
 )
 
 var (
 	v            = version.OperatorBinaryVersion
 	baseVersion  = v.MinorVersion.String()
 	patchVersion = v.PatchVersion.String()
+	buildArch    = buildversion.DockerInfo.Arch
+	buildOS      = buildversion.DockerInfo.OS
 
 	// ReleaseTar is a URL to download latest istio version from Github release
-	ReleaseTar = `https://github.com/istio/istio/releases/download/` + patchVersion + `/istio-` + patchVersion + `-linux-amd64.tar.gz`
+	ReleaseTar = fmt.Sprintf("https://github.com/istio/istio/releases/download/%s/istio-%s-%s-%s.tar.gz",
```

Comment:
I recommend that use string join to build a variable for `patchVersion`, `patchVersion`, `buildOS` and `buildArch`, then rend only one variable in `fmt.Sprintf`.